pkg/build/api/types.go

```diff
@@ -425,6 +428,19 @@ type SourceBuildStrategy struct {
 	ForcePull bool
 }
 
+// JenkinsPipelineStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
+	// JenkinsfilePath is the path of the Jenkinsfile that will be used to configure the pipeline
+	// relative to the root of the source repository.
+	JenkinsfilePath string `json:"jenkinsfilePath,omitempty"`
+
+	// Jenkinsfile is the raw contents of a Jenkinsfile containing the pipeline definition.
+	Jenkinsfile string `json:"jenkinsfile,omitempty"`
```

Comment:
Actually, after discussion I think I've been convinced that this belongs
with the strategy.  The reason the dockerfile was in the source was that we
assumed that multiple strategies could use it.  However, since a
Jenkinsfile is very strategy specific, I think it's ok to leave in the
strategy (it's like the FROM on a Docker build).

On Tue, Mar 22, 2016 at 5:20 AM, Jimmi Dyson <notifications@github.com>
wrote:

> In pkg/build/api/types.go
> <https://github.com/openshift/origin/pull/7949#discussion_r56955552>:
>
> > @@ -425,6 +428,19 @@ type SourceBuildStrategy struct {
> >  	ForcePull bool
> >  }
> >
> > +// JenkinsPipelineStrategy holds parameters specific to a Jenkins Pipeline build.
> > +type JenkinsPipelineBuildStrategy struct {
> > +	// JenkinsfilePath is the path of the Jenkinsfile that will be used to configure the pipeline
> > +	// relative to the root of the source repository.
> > +	JenkinsfilePath string `json:"jenkinsfilePath,omitempty"`
> > +
> > +	// Jenkinsfile is the raw contents of a Jenkinsfile containing the pipeline definition.
> > +	Jenkinsfile string `json:"jenkinsfile,omitempty"`
>
> Moving to source - should always have been there, sorry.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7949/files/969dde370a1a137068f3323edd5940f0374f9f0a#r56955552>
>